garak/generators/nim.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""NVIDIA Inference Microservice LLM interface"""
+
+import json
+import logging
+import os
+from typing import List, Union
+
+import requests
+
+import backoff
```

Comment:
is the `import openai` required? it feels like it would be nice if anything dependent on this 3rd-party module could be localised to garak.generators.openai (as far as possible), brought in when we import the mixin class

Reply:
Yes it is still required as the import in `generators/openai.py` does not bring the object into the local stack for this class to utilize in `_load_client()`:
```
Traceback (most recent call last):
  File "/Users/jemartin/.pyenv/versions/3.10.14/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/jemartin/.pyenv/versions/3.10.14/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/jemartin/Projects/nvidia/garak/garak/__main__.py", line 13, in <module>
    main()
  File "/Users/jemartin/Projects/nvidia/garak/garak/__main__.py", line 9, in main
    cli.main(sys.argv[1:])
  File "/Users/jemartin/Projects/nvidia/garak/garak/cli.py", line 451, in main
    generator = getattr(generator_mod, generator_class_name)(
  File "/Users/jemartin/Projects/nvidia/garak/garak/generators/openai.py", line 121, in __init__
    self._load_client()
  File "/Users/jemartin/Projects/nvidia/garak/garak/generators/nim.py", line 55, in _load_client
    self.client = openai.OpenAI(base_url=self.url, api_key=self.api_key)
NameError: name 'openai' is not defined. Did you mean: 'open'?
```

However since we still in the original process I believer this does not cause a re-import only a lookup of the imported symbol in the PythonVM.

Reply:
ok, maybe it's fine that using `OpenAICompatible` requires one to be cognisant that `openai` is being used